### PR TITLE
[SYCL] Add vec and marray support to known_identity type trait

### PR DESCRIPTION
**Pull request by aobolensk**
Add support for sycl::vec<type, N> and sycl::marray<type, N> in sycl::known_identity type trait.

### CODE REVIEW

File: sycl/test/basic_tests/known_identity.cpp [hunk 1]
```diff
@@ -93,6 +95,288 @@ template <typename T> void checkBoolKnownIdentity() {
...
+    res &= (a[i] == b[i]);
+  }
+  return res;
+}
+
+template <typename T, int Num>
+typename std::enable_if<!std::is_same<T, half>::value &&
+                            !std::is_same<T, float>::value &&
+                            !std::is_same<T, double>::value,
+                        void>::type
```
Comment (steffenlarsen) [inline sycl/test/basic_tests/known_identity.cpp:111]:
```suggestion
std::enable_if_t<!std::is_same_v<T, half> && !std::is_same_v<T, float> &&
                 !std::is_same_v<T, double>>
```
Minor stylistic suggestion. Now that we have C++17 for this test and we have `std::enable_if_t` in use in the rest of the changes, I like the brevity of the above code suggestion. 😄

**Review by steffenlarsen — APPROVED** [review]
Very nice!

File: sycl/test/basic_tests/known_identity.cpp (hunk 1 above)
Comment (steffenlarsen) [reply sycl/test/basic_tests/known_identity.cpp]: The same changes can be applied to the other tests.

Comment (aobolensk) [reply sycl/test/basic_tests/known_identity.cpp]: Done

Comment (steffenlarsen) [reply sycl/test/basic_tests/known_identity.cpp]: Thanks!

**Review by steffenlarsen — APPROVED** [review]

**Pull request merged**